lib/fourflusher/executable.rb

```diff
@@ -162,15 +162,18 @@ def self.capture_command(executable, command, capture: :merge)
     def self.popen3(bin, command, stdout, stderr)
       require 'open3'
       Open3.popen3(bin, *command) do |i, o, e, t|
-        reader(o, stdout)
-        reader(e, stderr)
+        stdout_thread = reader(o, stdout)
+        stderr_thread = reader(e, stderr)
         i.close
 
         status = t.value
 
         o.flush
         e.flush
-        sleep(0.01)
+
+        # wait for both threads to process the streams
+        stdout_thread.join
```

Comment:
do we care of setting a limit here? maybe not. I think it should be OK.

Reply:
Good point! Looking at `def self.reader(input, output)`, if the stream is not finishing, the thread will not finish.

A timer should catch errors/endless running of the underlying cmd call. This should make the code more robust. At least if we pick the right timeout :)

How does something like this sound to you?

```ruby
fail Fourflusher::Informative, "Executable not finished in time `#{cmd}`" unless stdout_thread.join(1)
fail Fourflusher::Informative, "Executable not finished in time `#{cmd}`" unless stderr_thread.join(1)
```

Open questions:

- Target branch?
- Should we create a constant for the timeout?